From 2d07ad148b90519dbad70de1a35591d7ee430ee2 Mon Sep 17 00:00:00 2001
From: AlexisWilke <alexis@m2osw.com>
Date: Fri, 22 Aug 2014 03:25:17 -0700
Subject: [PATCH] Update fixing test of filename to include extension.

See bug report https://sourceforge.net/p/unigw/discussion/647573/thread/f4ff021d/
by Mathiew Shroeter.
Added a test to ensure that filenames having two periods before the extension are now accepted.
Added a test to ensure that a filename ending with a period is forbidden.
Unfortuantely a filename ending with a space is caught earlier so I could not add that test.
---
 wpkg/libdebpackages/wpkgar_build.cpp      |  2 +-
 wpkg/tests/unittests/unittest_package.cpp | 35 ++++++++++++++++++++++++++++++-
 wpkg/tests/unittests/unittest_package.h   |  2 ++
 3 files changed, 37 insertions(+), 2 deletions(-)

diff --git a/wpkg/libdebpackages/wpkgar_build.cpp b/wpkg/libdebpackages/wpkgar_build.cpp
index b8ade38..ceba6a8 100644
--- a/wpkg/libdebpackages/wpkgar_build.cpp
+++ b/wpkg/libdebpackages/wpkgar_build.cpp
@@ -3354,7 +3354,7 @@ void wpkgar_build::build_deb(const wpkg_filename::uri_filename& dir_name)
                 // this is a logic error as it should not happen
                 throw std::logic_error("filename does not include at least one \"/\", it cannot be valid in wpkgar_build::build_deb().");
             }
-            const std::string file_basename(filename.basename());
+            const std::string file_basename(filename.segment(filename.segment_size() - 1));
             if(file_basename == ".." || file_basename == ".")
             {
                 // ignore this or parent directories
